packages/mongodb-chatbot-ui/src/useConversation.tsx

```diff
@@ -591,6 +595,22 @@ export function useConversation(params: UseConversationParams) {
     dispatch({ type: "setMessageMetadata", messageId, metadata });
   };
 
+  const updateMessageMetadata: ConversationActor["updateMessageMetadata"] =
+    async ({ messageId, metadata }) => {
+      if (!state.conversationId) {
+        console.error(`Cannot updateMessageMetadata without a conversationId`);
+        return;
+      }
+      dispatch({
+        type: "setMessageMetadata",
+        messageId,
+        metadata: {
+          ...state.messages.find((m) => m.id === messageId)?.metadata,
```

Comment:
```suggestion
          ...(state.messages.find((m) => m.id === messageId)?.metadata ?? {}),
```

i think you need this so spread operator doesn't break on undefined

Reply:
Spread operator supports undefined

`{ ...undefined } => {}`

Reply:
noted! i must have been hallucinating...